From 29a87380874b58794561618da443f3d2e6aba9cc Mon Sep 17 00:00:00 2001
From: Santosh Kr Thakur <santosh.krx.thakur@intel.com>
Date: Tue, 12 Jan 2016 14:38:33 +0530
Subject: [PATCH 4/4] Audio: codec: rt5670: Three button Headset Event
 Detection

It detects event from three button headset when buttons
are pressed.

Change-Id: Ie959f5b7425f55f2a2d7268646c2c07344e6636c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-14961
Signed-off-by: Santosh Kr Thakur <santosh.krx.thakur@intel.com>
Reviewed-on: https://android.intel.com:443/456786
---
 sound/soc/codecs/rt5670.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rt5670.c b/sound/soc/codecs/rt5670.c
index 7204c45..9f55af5 100644
--- a/sound/soc/codecs/rt5670.c
+++ b/sound/soc/codecs/rt5670.c
@@ -64,7 +64,9 @@ struct rt5670_init_reg {
 static struct rt5670_init_reg init_list[] = {
 	{ RT5670_DIG_MISC	, 0xc019 }, /* fa[0]=1, fa[3]=1'b MCLK det,
 						fa[15:14]=11'b for pdm */
+	{ RT5670_IL_CMD         , 0x0005 },
 	{ RT5670_IL_CMD2	, 0x0010 }, /* set Inline Command Window */
+	{ RT5670_IL_CMD3	, 0x0009 },
 	{ RT5670_A_JD_CTRL1     , 0x0001 }, /* set JD1 mode 1 (1 port) */
 	{ RT5670_PRIV_INDEX	, 0x0014 },
 	{ RT5670_PRIV_DATA	, 0x9a8a },
@@ -191,7 +193,7 @@ static const u16 rt5670_reg[RT5670_VENDOR_ID2 + 1] = {
 	[RT5670_SV_ZCD1] = 0x0809,
 	[RT5670_IL_CMD] = 0x0001,
 	[RT5670_IL_CMD2] = 0x0049,
-	[RT5670_IL_CMD3] = 0x0009,
+	[RT5670_IL_CMD3] = 0x0024,
 	[RT5670_DRC_HL_CTRL1] = 0x8000,
 	[RT5670_ADC_MONO_HP_CTRL1] = 0xb300,
 	[RT5670_ADC_STO2_HP_CTRL1] = 0xb300,
-- 
1.9.1

